persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java

```diff
@@ -163,40 +164,95 @@ public void start() {
       compactor.setIndex(index);
       startIndex();
       final AtomicLong maxSeqId = new AtomicLong(0);
-      if (index.isLoaded()) {
-         log.debug("Not building the index - loaded from persisted state");
-      } else if (configuration.purgeOnStartup()) {
-         log.debug("Not building the index - purge will be executed");
+      boolean migrateData = false;
+
+      if (!configuration.purgeOnStartup()) {
+         // we don't destroy the data on startup
+         // get the old files
+         FileProvider oldFileProvider = new FileProvider(getDataLocation(), configuration.openFilesLimit(), "");
+         if (oldFileProvider.hasFiles()) {
+            org.infinispan.util.logging.Log.PERSISTENCE.info("Migrating data new format...");
```

Comment:
We should use the statements you introduced in `Log.java`

Reply:
The code will be a lot cleaner if we statically import `PERSISTENCE` log.

Reply:
:+1: